roms_tools/utils.py

```diff
@@ -0,0 +1,172 @@
+from numbers import Integral
+
+import numpy as np
+import xarray as xr
+
+
+def partition(
+    ds: xr.Dataset, nx: int = 1, ny: int = 1
+) -> tuple[list[int], list[xr.Dataset]]:
+    """
+    Split a ROMS dataset up into nx by ny spatial tiles.
+    """
+
+    # TODO also check they are positive integers
+    if not isinstance(nx, Integral) or not isinstance(ny, Integral):
+        raise ValueError("nx and ny must be integers")
+
+    # 'eta_rho' and 'xi_rho' are always expected to be present
```

Comment:
I just remembered that Ulla was trying to partition her river input file but couldn't do it because that file is missing `eta_rho` and `xi_rho` (see discussion on slack today, https://cworthyworkspace.slack.com/archives/C06M8FCAYPN/p1724925409666779).

While ROMS-Tools is currently not producing a river input file (or anything without the dimensions `eta_rho` and `xi_rho`), we maybe want `partition` to work on more general files. In the extreme case, if none of the spatial dimensions are present, the dataset will just be copied x number of times, where x is the number of tiles.

Reply:
Done in https://github.com/CWorthy-ocean/roms-tools/pull/101/commits/50d1422284a995a92cbe2f0515ca4fdf7203a641